sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/GetterBasedSchemaProvider.java

```diff
@@ -185,6 +186,12 @@
               keyType,
               valueType,
               setterFactory);
+    } else if (type.getTypeName().equals(TypeName.DATETIME)) {
```

Comment:
I don't think this is the correct place to put this. If anything the conversion should happen inside Row instead.